components/script/dom/mutationobserver.rs

```diff
@@ -21,16 +25,16 @@ pub struct MutationObserver {
 }
 
 impl MutationObserver {
-    fn new(global: &Window, callback: MutationCallback) -> MutationObserver {
+    fn new(global: &Window, callback: Rc<MutationCallback>) -> MutationObserver {
         MutationObserver {
             reflector_: Reflector::new(),
             callback: callback,
         }
     }
 
-    pub fn Constructor(global: &Window, callback: Rc<MutationCallback>) -> Result<Root<MutationObserver>, Error> {
-        let observer = MutationObserver::new(global, Rc::deref(&callback));
-        ScriptThread::add_mutation_observer(&observer)
+    pub fn Constructor(global: &Window, callback: Rc<MutationCallback>) -> Fallible<Root<MutationObserver>> {
+        let observer = MutationObserver::new(global, callback);
+        ScriptThread::add_mutation_observer(Root::from_ref(&*observer))
```

Comment:
add_mutation_observer should accept a `&MutationObserver` argument instead of `Root<MutationObserver>`.